programs/rodeos/wasm_ql_http.cpp

```diff
@@ -445,12 +447,37 @@ class http_session {
          queue_(*this) {}
 
    // Start the session
-   void run() { do_read(); }
+   void run() { s data
+      _timer.reset(new boost::asio::steady_timer(derived_session().stream.socket().get_executor()));
+      session_begin = steady_clock::now();
+      _timer->expires_after( http_config->idle_timeout_ms );
+      _timer->async_wait(boost::asio::bind_executor(derived_session().stream.socket().get_executor(), boost::bind(&http_session::wait_to_timeout, derived_session().shared_from_this())));
```

Comment:
I don't see where this timer is cancelled? Shouldn't it be cancelled when a message comes in?

Reply:
Time gets cancelled through `wait_to_timeout` method which is called inside `boost::bind::executor()`